### PR TITLE
Use string/contains for new high cardinality string field filters

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/29881

For string fields with fingerprint info, we can default to `string/contains` if there are more than 20 unique values.

How to verify:
- New -> SQL query -> `SELECT * FROM PEOPLE WHERE {{filter}}`
- Change the type from `Text` to `Field Filter` -> `People` -> `Name`
- Do the same again with `People` -> `Source`. It should be just `String` because it's a low cardinality field.

Screenshot 2023-04-07 at 14 33 46

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29885)

### CODE REVIEW

**Comment by mazameli** [comment]
Looks good to me.

**Comment by ranquild** [comment]
@mazameli would it be ok to merge (& ship) this change without making `string contains` filter case-insensitive? or is it a blocker?

File: frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.unit.spec.tsx [hunk 1]
```diff
@@ -107,7 +107,25 @@ describe("TagEditorParam", () => {
...
+    it("should default to string/= for a new low cardinality string field filter", async () => {
+      const tag = createMockTemplateTag({
+        type: "dimension",
+        dimension: undefined,
+        "widget-type": undefined,
+      });
+      const { setTemplateTag } = setup({ tag });
+
+      userEvent.click(await screen.findByText("People"));
+      userEvent.click(await screen.findByText("Source"));
```
Comment (uladzimirdev) [inline frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.unit.spec.tsx:119]: nit: if you already waited for People, most likely Source is there already, so screen.getByText could work as well

**Review by uladzimirdev — APPROVED** [review]
From code perspective LGTM

File: frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.unit.spec.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.unit.spec.tsx:119]: ignore it please

**Pull request merged**